src/modules/csm/distribution.py

```diff
@@ -219,19 +240,25 @@ def process_validator_duties(
             #    87.55 ≈ 88 of 103 participation shares should be counted for the operator key's reward.
             #    The rest 15 participation shares should be counted for the protocol's rebate.
             #
-            participation_share = math.ceil(attestation.assigned * reward_share)
+            participation_share = math.ceil(duties.attestation.assigned * reward_share)
             participation_shares[validator.lido_id.operatorIndex] += participation_share
```

Comment:
What about to extract the `participation_share` part to the `ValidatorDutyOutcome` and do the modification of shares mapping at the outer level?